SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -268,7 +275,8 @@
 
   <!-- We want to override any properties that have been set declaratively in the project -->
   <Target Name="OverrideCodeAnalysisProperties" Condition=" $(SonarQubeTempPath) != '' "
-          BeforeTargets="RunCodeAnalysis;WriteSonarQubeProjectData" >
+          DependsOnTargets="SonarQubeCategoriseProject"
+          BeforeTargets="RunCodeAnalysis" >
```

Comment:
This target needs to know whether the project has been excluded or is a test project i.e. depends on the properties set in "SonarQubeCategoriseProject".